ReactAndroid/release.gradle

```diff
@@ -79,6 +79,7 @@ afterEvaluate { project ->
         classpath += files(project.getConfigurations().getByName("compile").asList())
         include("**/*.java")
         exclude("**/ReactBuildConfig.java")
+        exclude("**/ReactEditText.java")
```

Comment:
can you try removing these 2 lines then adding this in before line 80?

```
classpath += files("$buildDir/generated/source/buildConfig/release")
```

Not sure if there's a better way to get the buildConfig dir...

Reply:
Will try this early tomorrow. Do you know what gradle task generates the BuildConfig? I noticed we seem to run the JavaDoc generation task before some other parts of the build.

```
#!/bin/bash -eo pipefail
node ./scripts/publish-npm.js --nightly
Picked up _JAVA_OPTIONS: -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/6.7/userguide/gradle_daemon.html.
Daemon will be stopped at the end of the build stopping after processing
> Task :gradlePlugin-build:gradlePlugin:compileJava UP-TO-DATE
> Task :gradlePlugin-build:gradlePlugin:pluginDescriptors UP-TO-DATE
> Task :gradlePlugin-build:gradlePlugin:processResources UP-TO-DATE
> Task :gradlePlugin-build:gradlePlugin:classes UP-TO-DATE
> Task :gradlePlugin-build:gradlePlugin:jar UP-TO-DATE

> Task :ReactAndroid:androidJavadoc
```

Reply:
> Do you know what gradle task generates the BuildConfig?

That's just part of Gradle base functionality, explained here: https://developer.android.com/studio/build/gradle-tips#share-custom-fields-and-resource-values-with-your-app-code

Reply:
I tested this quickly, and I think generation doesn't happen before the task is run. The task succeeded when I had an already built project, but failed if I deleted `ReactAndroid/build`. first (though other bits were populated in the build directory).

I can dig around a bit more though. It seems like this is a problem that others would have run into.

Reply:
If you find out which task produces the BuildConfig, we can just make the javadoc task depend on that task

Reply:
Found it. Looks like I can declare `dependsOn:generateReleaseBuildConfig` and things work. The task seems somewhat private, so I wonder if it makes sense to do a more full Android build, but it solves the problem nicely.